src/frontend/planner_test/tests/testdata/batch_index_join.yaml

```diff
@@ -76,3 +76,15 @@
     └─BatchLookupJoin { type: Inner, predicate: t1.a = idx.c AND t1.b = idx.d, output: all }
       └─BatchExchange { order: [], dist: UpstreamHashShard(t1.a) }
         └─BatchScan { table: t1, columns: [t1.a, t1.b], distribution: SomeShard }
+- name: shouldn't be a lookup join
+  sql: |
+    create table t(a int, b int);
+    create index idx on t(a, b) distributed by (a);
+    select * from t t1 join t t2 on t1.b = t2.b;
```

Comment:
Should we add a case where it doesn't include `distributed by`? I think any non prefix eq condition should also not use lookup join.

Reply:
I think this case is what you need.